uptane/services/inventorydb.py

```diff
@@ -169,4 +84,122 @@ def scrub_filename(fname, expected_containing_dir):
   return abs_fname
 
 
+class InventoryDataBase():
+  """
+  Dictionary design:
+
+  {
+    'vehicle manifests':
+             {
+               'vehicle_id_1': [ <full vehicle manifests, in order of receipt> ],
+               ...
+             },
+    'ecu_manifests':
+             {
+               'ecu_serial': [ <full ECU manifests, in order of receipt>],
+               ...
+             }
+    'public_keys':
+             {
+               'vehicle_primaries':
+               {
+                 'vehicle_id_1': { ecu_id: 'primary_ecu_id',
+                                   public_key: <publickey for primary ecu>
+                                 },
+                 ...
+               },
+               'all_ecus':
+               {
+                 'ecu_serial_2': <publickey>,
+                 ...
+               }
+             }
+  }
+  """
+  # Inventory database dictionary
+  inventory_db = {'vehicle_manifests': {},
+                  'ecu_manifests': {},
+                  'public_keys': {
+                                  'vehicle_primaries': {},
+                                  'all_ecus': {}
+                                 }
+                 }
+
+  # Register ECU
+  def register_ecu(self, isPrimary, vin, ecu_serial, public_key):
+    if isPrimary:
+      vehicle_primary_ids = self.inventory_db['public_keys']['vehicle_primaries']
+      if (vehicle_primary_ids and (vin in vehicle_primary_ids)):
```

Comment:
On the `register_ecu` method as a whole:

Here's what I would have done instead, *just given proposed functionality*:
```python
  def register_ecu(self, is_primary, vin, ecu_serial, public_key):

    if is_primary:
      if overwrite or vin not in self.inventory_db['public_keys']['vehicle_primaries']:
        self.inventory_db['public_keys']['vehicle_primaries'][vin] = \
            {'ecu_id': ecu_serial, 'public_key': public_key}

    # Note that the remaining logic applies to all ECUs, Primary or not.
    if overwrite or ecu_serial not in self.inventory_db['public_keys']['all_ecus']:
      self.inventory_db['public_keys']['all_ecus'][ecu_serial] = public_key
```

The differences:
 - 'all_ecus' should include Primary ECUs, so they are added there.
 - It doesn't overwrite existing values.
   (May not have been clear that that was not desired default behavior.)
 - It's shorter (5 lines vs 14 lines)
 - It doesn't use unnecessary temp variables.
 - Variable casing/naming is consistent.



But maybe we want to overwrite optionally, so with that functionality added:
```python
  def register_ecu(self, is_primary, vin, ecu_serial, public_key, overwrite=False):

    if is_primary:
      if overwrite or vin not in self.inventory_db['public_keys']['vehicle_primaries']:
        self.inventory_db['public_keys']['vehicle_primaries'][vin] = \
            {'ecu_id': ecu_serial, 'public_key': public_key}

      else:
        raise uptane.Error('Vehicle ' + repr(vin) + ' already has a Primary '
            'ECU registered with a public key.')


    # Note that the remaining logic applies to all ECUs, Primary or not.
    if overwrite or ecu_serial not in self.inventory_db['public_keys']['all_ecus']:
      self.inventory_db['public_keys']['all_ecus'][ecu_serial] = public_key

    else:
      raise uptane.Error(
          'ECU ' + repr(ecu_serial) + ' already has a public key registered.')
```